modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/SearchAsYouTypeFieldMapper.java

```diff
@@ -147,23 +148,26 @@ public Builder maxShingleSize(int maxShingleSize) {
         }
 
         @Override
-        public SearchAsYouTypeFieldType fieldType() {
-            return (SearchAsYouTypeFieldType) this.fieldType;
+        public Builder docValues(boolean docValues) {
+            if (docValues) {
+                throw new IllegalArgumentException("mapper [" + name() + "] of type [search_as_you_type] does not support doc values");
```

Comment:
is this a new check or was it silently rejected before?

Reply:
It was silently ignored before - I think an error message is better, but maybe it should just issue a warning in 7.x?

Reply:
for all of these subtle changes, on one hand we could consider them bugfixes. but I am afraid that users may rely on us silently accepting wrong mappings and we should rather deprecate first and remove in 8.0, if possible